### PR TITLE
Replace start command with run  to avoid confusion

### PR DESCRIPTION
**Pull request by derekwbrown**
with service lifetime commands.  Commands still function (for now) with
a deprecation notice

### CODE REVIEW

File: cmd/agent/app/start.go [hunk 1]
```diff
@@ -51,27 +48,12 @@ import (
 var (
 	startCmd = &cobra.Command{
 		Use:   "start",
-		Short: "Start the Agent",
-		Long:  `Runs the agent in the foreground`,
+		Short: "(DEPRECATED) Use \"run\" instead to start the Agent",
```
Comment (JulienBalestra) [inline cmd/agent/app/start.go]: Cobra command contain a [deprecated](https://godoc.org/github.com/spf13/cobra#Command.Deprecated) string.
Is it useful in our case ?

**Comment by masci** [comment]
I'm 👎 for this change, I specially don't like the `kill` command which is inherently ambiguous in the *nix realm.

**Comment by olivielpeau** [comment]
After having discussed this, let's change the scope of the PR:

Let's not change `stop` to `kill`, it would probably be even more confusing. Let's only keep the change on `start` -> `run`.

The main reason to change anything is too avoid users mistakenly typing:

```sh
sudo datadog-agent start
```

instead of

```sh
sudo service datadog-agent start
```

and starting the agent in the foreground as `root` when their intention was to start the agent as a service (as the `dd-agent` user).

TODO:
- [ ] update `Dockerfiles/agent/s6-services/agent/run`
- [ ] update all the docs of the repo mentioning `agent start` to `agent run`
- [ ] update all our service definition files (systemd, upstart, launchd)
- [ ] say in the deprecation message that `start` will be removed from version `6.4` of the Agent, and plan work to actually remove it completely for that version

**Comment by derekwbrown** [comment]
Based on all the feedback, backed out the `kill` change; now just changes `start` to `run`

File: cmd/agent/app/start.go [hunk 2]
```diff
@@ -51,27 +48,12 @@ import (
...
+		Long:  `(DEPRECATED) Use \"run\" instead to runs the agent in the foreground.  Command will be removed in 6.4`,
```
Comment (truthbk) [inline cmd/agent/app/start.go]: small typo here: `runs` -> `run`

File: cmd/agent/app/start.go (hunk 1 above)
Comment (truthbk) [reply cmd/agent/app/start.go]: I think it would.

**Review by truthbk — COMMENTED** [review]
Looks almost good to go, just a couple of typos.

**Review by arbll — DISMISSED** [review]
LGTM, let's just fix the typos

**Comment by olivielpeau** [comment]
@derekwbrown let's complete the TODO items in my [earlier comment](https://github.com/DataDog/datadog-agent/pull/1615#issuecomment-384972845) before merging this (note that at the moment the PR hasn't updated the `launchd` service definition yet -- i.e. the service def for macOS)

File: releasenotes/notes/replcmds-3a95564a5f58197b.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+deprecations:
+  - |
+    Begin deprecating "Agent start".  It is being replaced by "run".  Command
```
Comment (olivielpeau) [inline releasenotes/notes/replcmds-3a95564a5f58197b.yaml]: nit: lowercase agent in "agent start" :)

Also, let's mention that it's the agent start _command_ in the 1st sentence

File: releasenotes/notes/replcmds-3a95564a5f58197b.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    will continue to function, with a deprecation notice
+other:
+  - |
+    Added new command "run" to the agent.  This command replacees the "start"
```
Comment (olivielpeau) [inline releasenotes/notes/replcmds-3a95564a5f58197b.yaml]: typo: `replaces`

**Comment by derekwbrown** [comment]
BOSH PR https://github.com/DataDog/datadog-agent-boshrelease/pull/38; should be merged after this is merged.

**Comment by xvello** [comment]
Docker image changes look good 👍

File: releasenotes/notes/replcmds-3a95564a5f58197b.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+deprecations:
+  - |
+    Begin deprecating "Agent start" command.  It is being replaced by "run".  Command
```
Comment (hush-hush) [inline releasenotes/notes/replcmds-3a95564a5f58197b.yaml]: I would replace `Command  will continue to function` by `The "start" command  will continue to function`, to be extra clear

File: cmd/agent/app/stop.go [hunk 6]
```diff
@@ -31,7 +31,7 @@ func init() {
 	AgentCmd.AddCommand(stopCmd)
 }
 
-func stop(*cobra.Command, []string) error {
+func stop(cmd *cobra.Command, args []string) error {
```
Comment (hush-hush) [inline cmd/agent/app/stop.go]: Why add back the variable ? Having the type with no variable is a common way among golang packages to signal that those parameters aren't used anywhere in the function.

**Review by hush-hush — COMMENTED** [review]
small nit-pick, seems good otherwise

File: cmd/agent/app/stop.go (hunk 6 above)
Comment (derekwbrown) [reply cmd/agent/app/stop.go]: leftover from the reverted `kill` command; removing.

**Review by hush-hush — DISMISSED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by hkaj — APPROVED** [review]

**Pull request merged**